File: backend/app/schemas/panchanga.py
from pydantic import BaseModel


class TithiResponse(BaseModel):
    number: int
    name: str
    paksha: str
    elapsed_pct: float


class VaraResponse(BaseModel):
    number: int
    english: str
    ruler: str
    sanskrit: str


class NakshatraResponse(BaseModel):
    number: int
    name: str
    pada: int
    elapsed_pct: float


class YogaResponse(BaseModel):
    number: int
    name: str
    elapsed_pct: float


class KaranaResponse(BaseModel):
    number: int
    name: str
    is_fixed: bool
    elapsed_pct: float


class PanchangaResponse(BaseModel):
    date: str
    local_time: str
    julian_day: float
    sun_longitude: float
    moon_longitude: float
    tithi: TithiResponse
    vara: VaraResponse
    nakshatra: NakshatraResponse
    yoga: YogaResponse
    karana: KaranaResponse

File: mobile/src/types/panchanga.ts
export interface TithiInfo {
  number: number;
  name: string;
  paksha: string;
  elapsed_pct: number;
}

export interface VaraInfo {
  number: number;
  english: string;
  ruler: string;
  sanskrit: string;
}

export interface NakshatraInfo {
  number: number;
  name: string;
  pada: number;
  elapsed_pct: number;
}

export interface YogaInfo {
  number: number;
  name: string;
  elapsed_pct: number;
}

export interface KaranaInfo {
  number: number;
  name: string;
  is_fixed: boolean;
  elapsed_pct: number;
}

export interface PanchangaData {
  date: string;
  local_time: string;
  julian_day: number;
  sun_longitude: number;
  moon_longitude: number;
  tithi: TithiInfo;
  vara: VaraInfo;
  nakshatra: NakshatraInfo;
  yoga: YogaInfo;
  karana: KaranaInfo;
}
